From 5b7183c77a2be9c3c26f9f209e783cb1a34118e0 Mon Sep 17 00:00:00 2001
From: Xaver Hugl <xaver.hugl@gmail.com>
Date: Thu, 7 Dec 2023 14:40:40 +0100
Subject: [PATCH] backends/drm: commit m_next state properly

Without this, atomic test failures might restore state that's out of date.

BUG: 477451
---
 src/backends/drm/drm_pipeline.cpp | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/src/backends/drm/drm_pipeline.cpp b/src/backends/drm/drm_pipeline.cpp
index e0e0b2f0195..84ee4d16557 100644
--- a/src/backends/drm/drm_pipeline.cpp
+++ b/src/backends/drm/drm_pipeline.cpp
@@ -433,13 +433,12 @@ void DrmPipeline::atomicCommitSuccessful()
             m_pending.crtc->cursorPlane()->commit();
         }
     }
-    m_current = m_pending;
+    m_current = m_next = m_pending;
 }
 
 void DrmPipeline::atomicModesetSuccessful()
 {
     atomicCommitSuccessful();
-    m_pending.needsModeset = false;
     if (activePending()) {
         pageFlipped(std::chrono::steady_clock::now().time_since_epoch());
     }
-- 
GitLab

